### PR TITLE
Simple Mileage-Only Claim in VAHB

### PR DESCRIPTION
**Pull request by juancstlm-a6**
## Link to Issue
https://github.com/department-of-veterans-affairs/va.gov-team/issues/91550

## Screenshots/Video
<details>
<summary><strong>📱 Toggle to View Mobile Screenshots Comparison Table</strong></summary>

<br>

| Screen Name                  | Android | iOS |
|-----------------------------|---------|-----|
| Address Screen              | Android - Address Screen | iOS - Address Screen |
| Burden Statement Screen     | Android - Burden Statement | iOS - Burden Statement |
| File Claim Alert            | Android - File Claim Alert | iOS - File Claim Alert |
| Interstitial Screen         | Android - Interstitial Screen | iOS - Interstitial Screen |
| Mileage Screen              | Android - Mileage Screen | iOS - Mileage Screen |
| Review Claim Screen         | Android - Review Claim | iOS - Review Claim |
| Review Claim Screen (Error) | Android - Review Claim Error | iOS - Review Claim Error |
| Submit Success Screen       | Android - Submit Success | iOS - Submit Success |
| Travel Agreement Screen     | Android - Travel Agreement | iOS - Travel Agreement |
| Travel Claim Help Screen    | Android - Claim Help | iOS - Claim Help |
| Vehicle Screen              | Android - Vehicle Screen | iOS - Vehicle Screen |
| Past Appointments List      |  |  |
| Appointment Details Screen  |  |  |
| Appointment in List         | Android - Appointment in List | iOS - Appointment in List |
| Error Screen - No Address   | Android - Error No Address | iOS - Error No Address |
| Error Screen - Unsupported Type | Android - Error Unsupported Type | iOS - Error Unsupported Type |

</details>

### Claim submission
- [ ] Users are able to submit a SMOC through the VAHB Mobile App
  - [ ] Travel Pay help is available on every screen
  - [ ] Answering "No" to any question displays the ineligible type screen
    - [ ] Tapping "close" on an error screen (ineligible type or no address) returns the user to the appointment details screen
  - [ ] Cannot submit until travel agreement checkbox is checked
  - [ ] After successful submission tapping "close" or "go to appointment" takes the user to the updated appointment details screen where there is now travel claim details component and no longer a "file travel claim" alert

### Appending travel claim data to appointments:
Appointments eligibility:
- Appointment is less than 30 days ago
- Only In person and Atlas appointments are eligible to file through the VAHB app at this time
- Other appointments may have existing claims filed through the existing BTSSS portal (e.g. community care)

- [ ] Travel Claim information is available in the appointment list and appointment details where applicable
  - [ ] **On appointment list:**
    - [ ] If existing claim is not present and meets eligibility requirements:
      - [ ] A blue pill with "[x] days to file travel claim" is present on appointment list item
  - [ ] **On appointment details screen:**
    - [ ] If Claim is present
      - [ ] Displays claim information (regardless of mobile app _filing_ eligibility)
    - [ ] If claim is not present:
      - [ ]  There are no errors retrieving claims and meets eligibility requirements:
        - [ ] File claim alert appears at the top of the appointment details screen
      - [ ]  There are no errors retrieving claims and appointment is MORE than 30 days old and meets eligibility requirements:
        - [ ] Messaging that a claim was not filed and there is a 30 day limit appears
      - [ ] There were errors appending claims to appointments:
        - [ ] Messaging that travel claims are unavailable with Help text is displayed

### Updating user address:
- [ ] If the user doesn't have an address they are routed to the "No Address" error screen when they tap yes on the vehicle question screen
- [ ] The "No Address" error screen will link the user to the place in the app where they can update their address right from the Travel Pay flow and would return them back to the flow after saving
- [ ] After saving their address the user will be brought back to the Travel pay flow on the address question screen

Test Users

- [ ] nollebarakat ([Link to credentials in team-sensitive repo](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/Administrative/vagov-users/staging-test-accounts-travel-pay.md))
- [ ] Demo Mode

## Checklist for PR Submitter

  **PR Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [x] Code is attached to a feature flag, or reason is given for no feature flag
- [x] AC and Tests are included to cover this change (when possible)
- [x] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [x] No secrets or API keys are checked in
- [x] All imports are absolute (no relative imports)
- [x] New functions and Redux work have proper TSDoc annotations

## Checklist for QA

  **QA Engineer:** Check off the items below as you test

Staging test user credentials available in the [vagov-team-sensitive repo here](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/Administrative/vagov-users/staging-test-accounts-travel-pay.md). (Some of the edge cases might be better tested in Demo mode.)

- [ ] Tested on iOS
- [ ] Tested on Android
- [x] [Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

**Review by liztownd — DISMISSED** [review]
A couple comments, but nothing major is jumping out at me.

**Review by liztownd — DISMISSED** [review]
🚀

**Comment by timwright12** [comment]
@juancstlm-a6 can you add a test user into the PR description for QA to use in testing this?

**Comment by liztownd** [comment]
> @juancstlm-a6 can you add a test user into the PR description for QA to use in testing this?

Sorry @timwright12  - I told Sam I'd add them but have been heads down on BE work. I'll add the link to the test user credentials shortly. Thanks!

File: VAMobile/src/screens/HealthScreen/Appointments/AppointmentTypeComponents/SharedComponents/AppointmentTravelClaimDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+const spacer = (theme: VATheme) => {
+  const boxProps: BoxProps = {
+    borderStyle: 'solid',
+    borderBottomWidth: 'default',
+    borderBottomColor: 'primary',
+    borderTopWidth: 'default',
+    borderTopColor: 'primary',
+    height: theme.dimensions.standardMarginBetween,
+    backgroundColor: 'main',
+    mx: -theme.dimensions.gutter,
```
[the remark this answers is not in the document]
Comment (juancstlm-a6) [reply VAMobile/src/screens/HealthScreen/Appointments/AppointmentTypeComponents/SharedComponents/AppointmentTravelClaimDetails.tsx:43]: I borrowed this pattern from the `AppointmentCancelReschedule` component for consistency, but let me see if there is better way.

File: VAMobile/src/screens/HealthScreen/Appointments/AppointmentTypeComponents/SharedComponents/AppointmentTravelClaimDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+function AppointmentTravelClaimDetails({ attributes, subType }: TravelClaimFiledDetailsProps) {
+  const { t } = useTranslation(NAMESPACE.COMMON)
+  const navigateTo = useRouteNavigation()
+  const theme = useTheme()
+
+  const travelPayInDowntime = useDowntime(DowntimeFeatureTypeConstants.travelPayFeatures)
+  const { downtimeWindowsByFeature } = useSelector<RootState, ErrorsState>((state) => state.errors)
+  const endTime =
+    downtimeWindowsByFeature[DowntimeFeatureTypeConstants.travelPayFeatures]?.endTime?.toFormat('EEEE, fff')
+
```
[the remark this answers is not in the document]
Comment (liztownd) [reply VAMobile/src/screens/HealthScreen/Appointments/AppointmentTypeComponents/SharedComponents/AppointmentTravelClaimDetails.tsx:57]: Also, for reference, here is the [DSVA Slack conversation](https://dsva.slack.com/archives/C08MEUZSG5V/p1747834417081309) around the `toFormat` token options as to why we went with the `('EEEE, FFF')` format. I believe there was going to be a brooder mobile ticket around adjusting other usages of that format elsewhere at some point.

[the remark this answers is not in the document]
Comment (jmarchi) [reply VAMobile/src/screens/HealthScreen/Appointments/AppointmentTypeComponents/SharedComponents/AppointmentTravelClaimDetails.tsx:57]: If this is something we need to use consistently can we add a util with a comment about it fixing the duplicate timezone render?

File: VAMobile/src/screens/HealthScreen/Appointments/AppointmentTypeComponents/SharedComponents/AppointmentTravelClaimDetails.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (dftony) [reply VAMobile/src/screens/HealthScreen/Appointments/AppointmentTypeComponents/SharedComponents/AppointmentTravelClaimDetails.tsx:43]: I saw it used in a couple places and was just curious but no big deal if there's no quick way to handle it

[the remark this answers is not in the document]
Comment (juancstlm-a6) [reply VAMobile/src/screens/HealthScreen/Appointments/AppointmentTypeComponents/SharedComponents/AppointmentTravelClaimDetails.tsx:43]: I think given that it's now being used in two places. A better approach is to make this a component itself. 
Ex: `<TextAreaSpacer/>` (not set on the name yet) to reduce code duplication

[the remark this answers is not in the document]
Comment (juancstlm-a6) [reply VAMobile/src/screens/HealthScreen/Appointments/AppointmentTypeComponents/SharedComponents/AppointmentTravelClaimDetails.tsx:43]: Created a shared component for the spacer https://github.com/department-of-veterans-affairs/va-mobile-app/pull/10990

File: VAMobile/src/screens/HealthScreen/Appointments/AppointmentTypeComponents/SharedComponents/AppointmentTravelClaimDetails.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (juancstlm-a6) [reply VAMobile/src/screens/HealthScreen/Appointments/AppointmentTypeComponents/SharedComponents/AppointmentTravelClaimDetails.tsx:57]: Added a util function for the formatting https://github.com/department-of-veterans-affairs/va-mobile-app/pull/10944/commits/5e0505283d389fe2560c50713b46cfb08d322edc

File: VAMobile/env/constant.env [hunk 3]
```diff
@@ -22,6 +22,10 @@ LINK_URL_APPLY_FOR_HEALTH_CARE=https://www.va.gov/health-care/apply/application/
 LINK_URL_HOW_TO_APPLY_FOR_HEALTH_CARE=https://www.va.gov/health-care/how-to-apply/
 LINK_URL_OMB_PAGE=https://www.reginfo.gov/public/do/PRAMain
 LINK_URL_MHV_VA_MEDICAL_RECORDS=https://www.va.gov/my-health/medical-records/
+LINK_URL_VA_FORM_10_3542="https://www.va.gov/find-forms/about-form-10-3542/"
+LINK_URL_TRAVEL_PAY_FILE_CLAIM_BTSSS="https://www.va.gov/health-care/get-reimbursed-for-travel-pay/#file-a-claim-for-general-healt"
```
Comment (dumathane) [inline VAMobile/env/constant.env:26]: 'healt' should probably be 'health'. These are also all quoted and the rest of the examples aren't, so can remove the quotes.

File: VAMobile/env/test.env.ts [hunk 4]
```diff
@@ -36,6 +36,14 @@ export const LINK_URL_GO_TO_VA_GOV = 'https://va.gov'
 export const LINK_URL_ABOUT_PACT_ACT = 'https://www.va.gov/resources/the-pact-act-and-your-va-benefits/'
 export const LINK_URL_APPLY_FOR_HEALTH_CARE = 'https://www.va.gov/health-care/apply/application/introduction'
 export const LINK_URL_HOW_TO_APPLY_FOR_HEALTH_CARE = 'https://www.va.gov/health-care/how-to-apply/'
+export const LINK_URL_VA_FORM_10_3542 = 'https://www.va.gov/resources/va-form-10-3542/'
+export const LINK_URL_TRAVEL_PAY_FILE_CLAIM_BTSSS =
+  'https://www.va.gov/health-care/get-reimbursed-for-travel-pay/#file-a-claim-for-general-healt'
```
[the remark this answers is not in the document]
Comment (liztownd) [reply VAMobile/env/test.env.ts:41]: Yeah, even with us spelling it with the `h` the link works, it just doesn't scroll to the section on the page we'd like it to. 🤷‍♀️

[the remark this answers is not in the document]
Comment (timwright12) [reply VAMobile/env/test.env.ts:41]: it might just be long and auto-truncated. Either way, no worries here

File: VAMobile/src/screens/HealthScreen/TravelPay/SubmitTravelPayFlowSteps/ReviewClaimScreen.test.tsx [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+    it('should show the loading screen', async () => {
+      mockIsPending = true
+      initializeTestInstance({ residentialAddress })
+      expect(screen.getByText(t('travelPay.submitLoading'))).toBeTruthy()
+      expect(screen.queryByTestId('reviewClaimScreenID')).toBeNull()
+    })
+  })
+
+  describe('when the submission is successful', () => {})
+
```
[the remark this answers is not in the document]
Comment (juancstlm-a6) [reply VAMobile/src/screens/HealthScreen/TravelPay/SubmitTravelPayFlowSteps/ReviewClaimScreen.test.tsx:144]: Looking back at this, it looks like this could use a couple of tests. The following two scenarios can be added here, although they are already tested in end-to-end (E2E) tests:

- It should verify that navigation routes to the next step.
- It should verify that the navigation parameters are correctly updated after success.

We can also remove this line and rely on E2E test for the above mentioned scenarios but I prefer to add them here as well

**Comment by juancstlm-a6** [comment]
> It looks like clicking the “Start a travel claim through BTSSS” link inside the Travel Claim Help leaves the Travel Claim Help open while the web view opens in the background. In order to access the webview the user will have to manually dismiss the Travel Claim Help by pressing close. Instead, should Travel Claim Help automatically be hidden as the webview is opened?
> 
> Below is a screenshot of the Travel Claim Help screen remaining open and blocking the webview after the link is pressed: 

@AdryienH fix for this in https://github.com/department-of-veterans-affairs/va-mobile-app/pull/10944/commits/7831302e8d41a16f22e5a075010ddc09d01d220e

File: VAMobile/env/test.env.ts [hunk 6]
```diff
@@ -36,6 +36,14 @@ export const LINK_URL_GO_TO_VA_GOV = 'https://va.gov'
 export const LINK_URL_ABOUT_PACT_ACT = 'https://www.va.gov/resources/the-pact-act-and-your-va-benefits/'
 export const LINK_URL_APPLY_FOR_HEALTH_CARE = 'https://www.va.gov/health-care/apply/application/introduction'
 export const LINK_URL_HOW_TO_APPLY_FOR_HEALTH_CARE = 'https://www.va.gov/health-care/how-to-apply/'
+export const LINK_URL_VA_FORM_10_3542 = 'https://www.va.gov/resources/va-form-10-3542/'
```
Comment (liztownd) [inline VAMobile/env/test.env.ts:39]: @dumathane pointed out that this url is incorrect and returns a 404. It should be updated to `https://www.va.gov/find-forms/about-form-10-3542/`. (It is correct in the `constant.env` file.) Since it's only incorrect in the test file we can make a point to update it in one of the follow up PRs.
